src/meta/api/src/background_api_impl.rs

```diff
@@ -0,0 +1,368 @@
+    }
+
+    async fn update_background_job_params(
+        &self,
+        req: UpdateBackgroundJobParamsReq,
+    ) -> Result<UpdateBackgroundJobReply, KVAppError> {
+        let name = &req.job_name;
+        let (id, mut resp) =
+            get_background_job_or_error(self, name, "failed to get background job").await?;
+        if resp.job_params == Some(req.params.clone()) {
+            return Ok(UpdateBackgroundJobReply { id });
+        }
+        resp.job_params = Some(req.params);
+        resp.last_updated = Some(Utc::now());
+        let req = UpdateBackgroundJobReq {
+            job_name: name.clone(),
+            info: resp,
+        };
+        self.update_background_job(req).await
+    }
+
+    async fn get_background_job(
+        &self,
+        req: GetBackgroundJobReq,
+    ) -> Result<GetBackgroundJobReply, KVAppError> {
+        debug!(req = debug(&req), "BackgroundApi: {}", func_name!());
+
+        let name_key = &req.name;
+
+        let (id, job) =
+            get_background_job_or_error(self, name_key, format!("get_: {:?}", name_key)).await?;
+
+        Ok(GetBackgroundJobReply { id, info: job })
+    }
+
+    async fn list_background_jobs(
+        &self,
+        req: ListBackgroundJobsReq,
+    ) -> Result<Vec<(u64, BackgroundJobInfo)>, KVAppError> {
+        let prefix = format!("{}/{}", BackgroundJobIdent::PREFIX, req.tenant);
+        let reply = self.prefix_list_kv(&prefix).await?;
+        let mut res = vec![];
+        for (_, v) in reply {
+            let job_id = deserialize_u64(&v.data)?;
+            let r = get_background_job_by_id(self, &BackgroundJobId { id: job_id.0 }).await?;
+            // filter none and get the task info
+            if let Some(task_info) = r.1 {
+                res.push((r.0, task_info));
+            }
+        }
+        Ok(res)
+    }
+
+    async fn update_background_task(
+        &self,
+        req: UpdateBackgroundTaskReq,
+    ) -> Result<UpdateBackgroundTaskReply, KVAppError> {
+        debug!(req = debug(&req), "BackgroundApi: {}", func_name!());
+        let name_key = &req.task_name;
+        let ctx = &func_name!();
+        let mut trials = txn_trials(None, ctx);
+        loop {
+            trials.next().unwrap()?;
+            debug!(?name_key, "update_background_task");
+
+            let meta = req.task_info.clone();
+            let if_then = vec![txn_op_put_with_expire(
+                name_key,
+                serialize_struct(&meta)?,
+                req.expire_at,
+            )];
+            let txn_req = TxnRequest {
+                condition: vec![],
+                if_then,
+                else_then: vec![],
+            };
+            let (succ, _responses) = send_txn(self, txn_req).await?;
+            if succ {
+                break Ok(UpdateBackgroundTaskReply {
+                    last_updated: meta.last_updated.unwrap_or(Default::default()),
+                    expire_at: req.expire_at,
+                });
+            }
```

Comment:
Is it necessary for the target to exist when performing an `update` operation?

And it looks updating an task is quite simple and can be done with a `KVApi::upsert_kv()`.  There is no need to run a transaction. Refer to a sample usage of `upsert_kv`:

https://github.com/datafuselabs/databend/blob/599d5b0841a2b17d59c265cc0ebe3e525b7f34ce/src/query/management/src/cluster/cluster_mgr.rs#L90-L93

Metadata operation that involves only one record, such as `delete_background_task()`, can always be done with `KVApi::upsert_kv()`, for example:
```rust
self.upsert_kv(UpsertKVReq::new(&key, seq, Operation::Delete, None))
```

Reply:
It is not necessary, since it assume to be a simple operation mainly used for audit purpose